bin/alluxio

```diff
@@ -46,77 +46,92 @@ function bootstrapConf {
     TOTAL_MEM=$(sysctl hw.memsize | cut -d ' ' -f2)
-    TOTAL_MEM=$[TOTAL_MEM / 1024]
+    TOTAL_MEM=$[TOTAL_MEM / 1024 / 1024]
     TOTAL_MEM=$[TOTAL_MEM * 2 / 3]
+    RAMDISK_FOLDER="/Volumes/ramdisk"
   else
-    ALLUXIO_SED="sed -i"
+    # Assuming Linux
     TOTAL_MEM=$(awk '/MemTotal/{print $2}' /proc/meminfo)
-    TOTAL_MEM=$[TOTAL_MEM * 2 / 3]
+    TOTAL_MEM=$[TOTAL_MEM / 1024 * 2 / 3]
+    RAMDISK_FOLDER="/mnt/ramdisk"
   fi
 
-  # Create a default config that can be overridden later
-  cp "${ALLUXIO_CONF_DIR}/alluxio-env.sh.template" ${dst}
-
   # Set the master hostname
-  ${ALLUXIO_SED} "s/XXX_MASTER_ADDRESS_XXX/$1/" ${dst}
+  ALLUXIO_MASTER_ADDRESS=${master}
 
   # Set the worker memory size
-  ${ALLUXIO_SED} "s/XXX_MEMORY_SIZE_XXX/${TOTAL_MEM}KB/" ${dst}
+  ALLUXIO_WORKER_MEMORY_SIZE="${TOTAL_MEM}MB"
+
+  # Set the ramdisk folder
+  ALLUXIO_RAM_FOLDER="${RAMDISK_FOLDER}"
 
   # Set under storage specific configuration
   case ${ufs} in
     local)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/\$\{ALLUXIO_HOME\}\/underFSStorage/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="\${ALLUXIO_HOME}/underFSStorage/"
       ;;
     hdfs)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/hdfs:\/\/localhost:9000\/alluxio/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="hdfs://localhost:9000/alluxio/"
       ;;
     s3)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/s3n:\/\/bucket\/directory/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="s3n://bucket/directory/"
       ;;
     gcs)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/gs:\/\/bucket\/directory/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="gs://bucket/directory/"
       ;;
     glusterfs)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/glusterfs:\/\/\//" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
-      export ALLUXIO_UNDERFS_GLUSTER_VOLUMES=\$\{ALLUXIO_UNDERFS_GLUSTER_VOLUMES:-alluxio_vol\}\\
-      " ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
-      export ALLUXIO_UNDERFS_GLUSTER_MOUNTS=\${ALLUXIO_UNDERFS_GLUSTER_MOUNTS:-\/vol\}\\
-      " ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
-      export ALLUXIO_UNDERFS_GLUSTER_MR_DIR=\${ALLUXIO_UNDERFS_GLUSTER_MR_DIR:-glusterfs:\/\/\/mapred\/system\}\\
-      " ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="/gluster_vol/"
       ;;
     swift)
-      ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/swift:\/\/swift-container/" ${dst}
-      ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+      ALLUXIO_UNDERFS_ADDRESS="swift://swift-container/"
       ;;
     *)
       rm ${dst}
       echo ${usage}
       exit 1
       ;;
   esac
+
+  cat ${src} | while read -r line
+  do
+    case ${line} in
```

Comment:
Could we avoid depending on the text of the template file in the bootstrap implementation? Also, what happened to generating the alluxio-site.properties file?